### DISCUSSION TITLE
Gosund SP1 2200W OTA flash fails

### DISCUSSION THREAD

**Discussion by Hervshahn** [ct-Open-Source/tuya-convert#1048] [General]
Hi,

I am a new member here and trying to get familiar with Tasmota and tuya-convert. I bought some Gosund SP1 plugs recently, 2 with 3680W and 2 with 2200W. Trying to flash Tasmota OTA with tuya-convert fails on the 2200W plugs. Are the 2200W plugs incompatible with Tasmota? The 3680W plugs run Tasmota without problems.

thanks for any kind of feedback :-)

**Comment by MacSass** [comment]
We would need to know what error you get.
Generally there should not be a difference - but it might be that the 2200W plugs e.g. have a newer firmware or they might even have a completely different chip (which would make them incompatible). 
So - more details from you are needed to help you ...

**Comment by Hervshahn** [comment]
Hi and thanks for the fast reply!

I don't know which chip is used on the SP1, I opened it but the chip is covered with a silver plate :-(  
The revision printed on the board is HYS-01-007 V3.0 2017-12-06 (if that helps?)

I installed tuya-convert on a Raspberry PI and started start_flash.sh, connected my Smartphone to  vrust-flash, pressed the button on the Gosund SP1 until the LED started flashing rapidly and pressed Enter on the Raspi. Shortly after having started this the SP1 klicks (switches) one time, and then returns to a slowly flashing LED stats.

The Raspi provided the following output:

```
Starting smart config pairing procedure
Waiting for the device to install the intermediate firmware
Put device in EZ config mode (blinking fast)
Sending SSID                  vtrust-flash
Sending wifiPassword
Sending token                 00000000
Sending secret                0101
................
SmartConfig complete.
Auto retry in 1s. ...................................................
Timed out while waiting for the device to (re)connect

Attempting to diagnose the issue...
An ESP82xx based device connected according to your wifi log.
If this is the device you are trying to flash, another issue may be preventing it from flashing.
Otherwise, it could be that the device does not use an ESP82xx or it did not connect.
For additional information, check the *.log files inside the scripts folder.
Please include these logs when opening a new issue on our GitHub issue tracker.
```

**Comment by MacSass** [comment]
OK,
the good news - looks like you not have an issue with the FW being too new - otherwise you would see that in the output of tuya-convert - so your device should usually be flashable :-)

The bad news:
I have the same issue with no solution for it yet. I had 5 identical plugs - two of them I could flash, for 3 of them I do have the same issue like you are seeing. Others have the same issue too and I don´t think anybody has been able to pinpoint the issue so far to the root cause.

Here are open issues that are similar - but different people seem to find different solutions, but none of them worked for me so far:
https://github.com/ct-Open-Source/tuya-convert/issues/948

I will try this next:
https://github.com/ct-Open-Source/tuya-convert/pull/966#issuecomment-1126897698

Unfortunately I can not give you good advise what to do, besides looking at the issues and trying ...

**Comment by Hervshahn** [comment]
Hmm - ok thanks for the feedback. I will try a serial flash with a USB-TTL adapter
